### PR TITLE
Add PHP 8.1 type hints to ArrayAccess and JsonSerializable

### PR DESCRIPTION
**Pull request by neildaniels**
PHP 8.1 added type hints to `ArrayAccess` and `JsonSerializable` functions. This PR adds those hints to avoid incompatible signature errors.

### CODE REVIEW

**Comment by neildaniels** [comment]
I tempered the type hint changes (namely, avoiding the use of `mixed`) to remain backwards compatible.

**Review by Ahummeling — APPROVED** [review]
Adding return types to non-final methods (i.e. abstract classes) would break BC for any extending classes that override these methods. Although, I cannot imagine why anyone would that with these methods tbh.
So if I understand correctly, you've used the `#[\ReturnTypeWillChange]` attribute for any method that would otherwise return `mixed`?

**Comment by neildaniels** [comment]
That's correct in regards to backwards compatibility. However, as of PHP 8.1 the classes specific to this project no longer conform to the `ArrayAccess` or `JsonSerializable` because they now officially have real return types.

So, yes, any extended classes would need to be updated. But, that'll be necessary at some point anyways.

- - -

Yes, the `#[\ReturnTypeWillChange]` attribute indicates something that _should_ return `mixed`. However, PHP 7 doesn't support a `mixed` return type, so this code would completely fail in that environment.

When PHP 7 support for this project is dropped, the attribute should be removed and a real return type of `mixed` should be used.

**Comment by Gummibeer** [comment]
Hey, I would more happily drop PHP7 here instead of adding a fix for an outdated PHP version which would require us to change it again at a later point. PHP7 is EOL so no reason to keep it in mind for changes.

**Comment by neildaniels** [comment]
I've gone ahead added the `mixed` return type and bumped the `composer.json`'s minimum PHP version to 8.0.

**Comment by Gummibeer** [comment]
https://github.com/spatie/schema-org/blob/main/.github/workflows/run-tests.yml
has to be updated to don't run PHP7.4 anymore.

**Comment by mziraki** [comment]
@Gummibeer @Ahummeling any update about this PR?!

**Comment by Ahummeling** [comment]
These changes look good to me, but I'm not a maintainer, so I cannot merge this.

**Comment by Gummibeer** [comment]
https://github.com/spatie/schema-org/releases/tag/3.10.0

**Pull request merged**
